### ISSUE TITLE
Error while creating and updating aws_ses_receipt_rule_set

### ISSUE THREAD

**Issue by janfoe** [terraform-providers/terraform-provider-aws#6470]
### Terraform Version

Terraform v0.11.10

- provider.aws v1.45.0

### Terraform Configuration Files

```hcl
resource "aws_ses_domain_identity" "domain" {
    provider = "aws.west"
    domain = "test.service.test.cloud"
}

resource "aws_ses_receipt_rule_set" "store_s3_ruleset" {
    provider = "aws.west"
    rule_set_name = "store_s3_ruleset"
}

resource "aws_ses_receipt_rule" "store_s3_rule" {
    provider    = "aws.west"
    name          = "store_s3_rule"
    rule_set_name = "store_s3_ruleset"
    enabled       = true
    scan_enabled  = true
    depends_on    = ["aws_ses_receipt_rule_set.store_s3_ruleset"]
    
    bounce_action {
      message = "Mailbox does not exist"
      smtp_reply_code = "550"
      status_code = "5.1.1"
      sender = "notavailable@test.cloud" 
      position = "1"
    }
}
```

```hcl
provider "aws" {
  region = "eu-central-1"
}

provider "aws" {
  alias   = "west"
  region  = "eu-west-1"
}
```

```hcl
terraform {
  backend "s3" {
    bucket = "terraform-dev-test-ses"
    key    = "state/terraformsestest"
    region = "eu-central-1"
  }
}

data "terraform_remote_state" "app" {
  backend = "s3"

  config {
    bucket = "terraform-dev-test-ses"
    key    = "state/terraformsestest"

    region = "eu-central-1"
  }
}
```

### Debug Output
First 'terraform apply' works as expected and creates several resources:
```hcl
  + aws_ses_domain_identity.domain
      id:                                      <computed>
      arn:                                     <computed>
      domain:                                  "test.service.test.cloud"
      verification_token:                      <computed>

  + aws_ses_receipt_rule.store_s3_rule
      id:                                      <computed>
      bounce_action.#:                         "1"
      bounce_action.615010230.message:         "Mailbox does not exist"
      bounce_action.615010230.position:        "1"
      bounce_action.615010230.sender:          "notavailable@test.cloud"
      bounce_action.615010230.smtp_reply_code: "550"
      bounce_action.615010230.status_code:     "5.1.1"
      bounce_action.615010230.topic_arn:       ""
      enabled:                                 "true"
      name:                                    "store_s3_rule"
      rule_set_name:                           "store_s3_ruleset"
      scan_enabled:                            "true"
      tls_policy:                              <computed>

  + aws_ses_receipt_rule_set.store_s3_ruleset
      id:                                      <computed>
      rule_set_name:                           "store_s3_ruleset"
```

A second 'terraform apply' tries to create ruleset again.
```hcl
  + aws_ses_receipt_rule_set.store_s3_ruleset
      id:            <computed>
      rule_set_name: "store_s3_ruleset"
```

```hcl
Error: Error applying plan:

1 error(s) occurred:

* aws_ses_receipt_rule_set.store_s3_ruleset: 1 error(s) occurred:

* aws_ses_receipt_rule_set.store_s3_ruleset: Error creating SES rule set: AlreadyExists: Rule set already exists: store_s3_ruleset
        status code: 400
```

### Actual Behavior

I'm trying to create a ruleset including a bounce-rule in AWS Simple Email Service. When I call 'terraform apply' the first time, it behaves as expected and creates the rule set and the rule in AWS Simple Email Service. 
When I call 'terraform apply' a second time, Terraform seems trying to create the rule set again and reports the following error message: ''Error creating SES rule set: AlreadyExists: Rule set already exists: store_s3_ruleset"

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

**Comment by jeohist** [comment]
I get the same issue. This does not occur here when the aws_ses_receipt_rule_set was created in a previous version.

**Comment by brunzefb** [comment]
Its a timing issue with the aws_ses_receipt_rule executing before the s3 bucket/policy has finished updating.  My workaround is using a null-resource with a delay.

```
resource "aws_s3_bucket" "email_bucket" {
  bucket = "${var.bucketname}"
  acl    = "private"
  region = "${var.region}"
  versioning {
    enabled = true
  }
}

data "aws_caller_identity" "current" {}


resource "aws_s3_bucket_policy" "s3_ses" {
  bucket = "${var.bucketname}"

  policy = <<EOF
{
  "Id": "Policy1543436450657",
  "Version": "2012-10-17",
  "Statement": [
      {
          "Sid": "AllowSESPuts",
          "Effect": "Allow",
          "Principal": {
              "Service": "ses.amazonaws.com"
          },
          "Action": "s3:*",
          "Resource": "arn:aws:s3:::${var.bucketname}/*",
          "Condition": {
              "StringEquals": {
                  "aws:Referer": "${data.aws_caller_identity.current.account_id}"
              }
          }
      }
  ]
}
EOF
  depends_on = ["aws_s3_bucket.email_bucket"]
}

# Add a header to the email and store it in S3
resource "aws_ses_receipt_rule" "store" {
  name          = "store"
  rule_set_name = "default-rule-set"
  recipients    = ["admin@${var.domain}"]
  enabled       = true
  scan_enabled  = true

  s3_action {
    bucket_name = "${var.bucketname}"
    position    = "1"
  }

  depends_on = ["null_resource.delay"]
}

# this is a trick to inject a delay between bucket/policy creation
# and creation of the aws_ses_receipt_rule
resource "null_resource" "delay" {
  provisioner "local-exec" {
    command = "sleep 10"
  }
  triggers = {
    "before" = "${aws_s3_bucket_policy.s3_ses.id}"
  }
}
```

**Comment by janfoe** [comment]
Thanks for your answer @brunzefb , but that does not fix my problem. 
As you can see in my example, I do not use S3 action, but only perform a bounce action.

**Comment by bflad** [comment]
Pull request submitted for the original issue here: #9045

**Comment by bflad** [comment]
The expected fix for this has been merged and will release with version 2.16.0 of the Terraform AWS Provider, likely tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 2.16.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2160-june-20-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
